client/driver/docker.go

```diff
@@ -471,7 +480,14 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
 				if authConfigurations, err = docker.NewAuthConfigurations(f); err != nil {
 					return nil, fmt.Errorf("Failed to create docker auth object: %v", err)
 				}
-				if authConfiguration, ok := authConfigurations.Configs[repo]; ok {
+
+				authConfigurationKey := ""
+				if driverConfig.SSL {
+					authConfigurationKey += "https://"
+				}
+
+				authConfigurationKey += strings.Split(driverConfig.ImageName, "/")[0]
```

Comment:
No, the split command returns the image string with the url scheme and without the url path -- which is different from the repo string.